### PR TITLE
upgrade spark version

### PR DESCRIPTION
**Pull request by benmayne**

### CODE REVIEW

**Comment by benmayne** [comment]
@ndigati can you set this branch up as a dockerhub tag for a bit to play with?

**Comment by nfultz** [comment]
Is there a corresponding edit for spark configs? It would be nice to get rid of 

```
18/07/31 17:32:26 WARN spark.SparkConf: The configuration key 'spark.yarn.access.namenodes' has been deprecated as of Spark 2.2 and may be removed in the future. Please use the new key 'spark.yarn.access.hadoopFileSystems' instead.
```

**Comment by benmayne** [comment]
yeah, I can probably swing that. Trying to sort this out right now https://stackoverflow.com/questions/49835433/java-lang-classcastexception-org-apache-hadoop-conf-configuration-cannot-be-cas

**Comment by ndigati** [comment]
@benmayne new branch tag is building

**Comment by ndigati** [comment]
`factual/docker-cdh5-dev:feature_spark-2.3.1` just finished building @benmayne

File: Dockerfile [hunk 1]
```diff
@@ -65,14 +65,17 @@ RUN curl -fsSL https://raw.githubusercontent.com/Factual/drake/master/bin/drake
...
+ENV SPARK_VERSION=2.3.1
 ENV SPARK_HOME=/opt/spark
+
 RUN mkdir -p $SPARK_HOME \
   && curl -fsSL http://archive.apache.org/dist/spark/spark-$SPARK_VERSION/spark-$SPARK_VERSION-bin-hadoop2.6.tgz \
     | tar xz -C /tmp/ \
   && rm -rf $SPARK_HOME && mv /tmp/spark-$SPARK_VERSION-bin-hadoop2.6 $SPARK_HOME \
   && mkdir -p /etc/spark/ \
   && ln -s $SPARK_HOME/conf /etc/spark/conf
+ADD spark-defaults.conf /etc/spark/conf/
```
Comment (ndigati) [inline Dockerfile]: Minor thing but can we change this `ADD` to `COPY`. `COPY` is generally preferred over `ADD` when your dealing with normal (non-tar) files.

Comment (benmayne) [reply Dockerfile]: yeah

**Pull request merged**